modules/utilities.py

```diff
@@ -0,0 +1,243 @@
+
+# ----- KERAS -----
+import tensorflow.keras as keras
+from keras.models import Sequential
+# ----- KERAS -----
+
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+import seaborn as sns
+from sklearn.metrics import confusion_matrix
+
+from sklearn.metrics import roc_curve, auc
+
+
+def get_imdb_reviews(path_to_directory=None):
+    """
+        Test function fixed into iMDB example
+    :param path_to_directory: (str) path to directory of reviews
+    :return: (list) list of reviews
+    """
+    list_of_reviews = []
+
+    for path in pathlib.Path(path_to_directory).iterdir():
+        if path.is_file():
+            current_file = open(path, "r")
+            list_of_reviews.append(current_file.read())
+            current_file.close()
+
+    return list_of_reviews
+
+
+def import_dataset(path_to_file=None):
+    """
+        Import the dataset based on the path to file URL
+
+    :param path_to_file: (str) path to CSV file
+    :return: (pandas.Dataframe) Pandas dataframe containing data
+    """
+    return pd.read_csv(path_to_file)
+
+
+def clean_text(text):
+    """
+        Utility function to help with cleaning up raw-input text. The following is performed:
+            - filtering out punctuations
+            - transform text to lowercase
+            - reduce words to their roots
+
+    :param text: (str) input raw-text to be processed
+    :return: (str) processed text
+    """
+    lemmatizer = WordNetLemmatizer()
+    stop_words = set(stopwords.words("english"))
+
+    text = re.sub(r'[^\w\s]', '', str(text))
+    text = text.lower()
+    text = [lemmatizer.lemmatize(token) for token in text.split(" ")]
+    text = [lemmatizer.lemmatize(token, "v") for token in text]
+    text = [word for word in text if not word in stop_words]
+    text = " ".join(text)
+    return text
+
+
+def build_bidirectional_lstm(N, max_features, embed_size, is_attention=False):
+    """
+        Builder function for the Bidirectional LSTM with user-specification of an Attention Layer.
+
+    :param N: (int) input length
+    :param max_features: (int) maximum number of features
+    :param embed_size: (int) size of embedding
+    :return: (tf.keras.Model) LSTM model
+    """
+    if is_attention:
+        sequence_input = Input(shape=(N,))
+        embedded_sequences = Embedding(max_features, embed_size)(sequence_input)
+
+        (bilstm, forward_h, forward_c, backward_h, backward_c) = Bidirectional(
+            LSTM(196,
+                 dropout=0.2,
+                 return_state=True,
+                 return_sequences=True)
+        )(embedded_sequences)
+
+        state_h = Concatenate()([forward_h, backward_h])
+        state_c = Concatenate()([forward_c, backward_c])
+
+        context_vector, attention_weights = Attention(10)(bilstm, state_h)
+
+        dense1 = Dense(20, activation="relu")(context_vector)
+
+        dropout = Dropout(0.05)(dense1)
+        output = Dense(1, activation="sigmoid")(dropout)
```

Comment:
This is where I was thinking about swapping out the 1-node output layer from Sentiment analysis and replacing it with an N-node output layer for topic classification (i.e. `Dense(N, activation="softmax")(dropout)`)

where N is the list of pre-defined topic tags that Gawtham showed us in past meetings.